lib/rex/socket/ssl.rb

```diff
@@ -0,0 +1,148 @@
+# -*- coding: binary -*-
+require 'rex/socket/x509_certificate'
+require 'timeout'
+
+###
+#
+# This class provides methods for interacting with an SSL wrapped TCP server.  It
+# implements the StreamServer IO interface.
+#
+###
+module Rex::Socket::Ssl
+
+  @@loaded_openssl = false
+
```

Comment:
do you think we really need to continue pretending openssl is optional? I don't think Ruby even compiles without it anymore. I'd suggest removing this entire @@loaded_openssl path

Reply:
Done sir done.